src/SvgUtils.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2023 Inclusive Design Research Centre, OCAD University
+ * All rights reserved.
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/inclusive-design/adaptive-palette/blob/main/LICENSE
+ */
+
+import { BlissSVGBuilder } from "bliss-svg-builder";
+import { BciAvId } from "./BlissSymbol";
+import { adaptivePaletteGlobals } from "./GlobalData";
+
+export const DEFAULT_SVG_MARKUP_STRING = "B3";   // Question mark
+
+/**
+ * Convert the given `BciAvId` to a SVG builder code string.  If the `BciAvId`
+ * argument is an array of BCI-AV-IDs and punctuation, concatenate the array
+ * into a string of builder code strings and punctuation marks.
+ * @param {BciAvId} - The BciAvId to convert.
+ * @return {String} - The concatenation of the builder codes and punctuation,
+ *                    e.g., "B106/B12".
+ */
+export function bciAvIdToString (bciAvId: BciAvId) {
+  let finalCode = "";
+  if (typeof bciAvId === "number") {
+    const { blissSvgBuilderCode } = bciToBlissaryId(bciAvId);
+    finalCode = blissSvgBuilderCode;
+  }
+  // `bicAvId` is an array
+  else {
+    bciAvId.forEach((item) => {
+      if (typeof item === "number") {
+        const { blissSvgBuilderCode } = bciToBlissaryId(item);
+        finalCode = `${finalCode}${blissSvgBuilderCode}`;
+      } else {
+        finalCode = `${finalCode}${item}`;
+      }
+    });
+  }
+  return finalCode;
+}
+
+/**
+ * Get the SVG markup as a string based on the given single BCI-AV-ID.
+ * or an array of BCI-AV-IDs and other characters
+ *
+ * @param {BciAvId} bciAvId - A single BCI-AV-ID (a number) or an array of such
+ *                            ids and characters, e.g. `[ 12335, "/", 8499 ]`
+ * @return {String} - The corresponding SVG markup, or the empty string.
+ */
+export function getSvgMarkupString (bciAvId: BciAvId) {
+  let builder;
+  try {
+    const svgBuilderArgument = bciAvIdToString(bciAvId);
+    builder = new BlissSVGBuilder(svgBuilderArgument);
+  }
+  catch (err) {
+    console.error(err);
+    console.error(`GETSVGMARKUPSTRING(): using question mark for SVG builder argument from bci-av-id = ${bciAvId}`);
+    builder = new BlissSVGBuilder(DEFAULT_SVG_MARKUP_STRING); // question mark
```

Comment:
I'm thinking when an error occurs and the SVG builder doesn't return a SVG, would it make more sense to return `undefined` instead of a SVG of a question mark? Returning `undefined` can be captured by the caller component to deal as an exception. Returning a question mark seems like a silent fail that the caller would assume things went well and a question mark is the expected.

Reply:
I agree this is a design issue. At this development stage, I lean towards the option 1 that does not render any SVG, as well as reporting a `console.error()` with what's happening so developers realize this BCI-AV-ID should be checked.